vpr/src/route/rr_graph.cpp

```diff
@@ -3060,11 +3128,13 @@ static vtr::NdMatrix<std::vector<int>, 4> alloc_and_load_pin_to_track_map(const
     /* allocate 'result' matrix and initialize entries to OPEN. also allocate and intialize matrix which will be
      * used to index into the correct entries when loading up 'result' */
 
-    auto result = vtr::NdMatrix<std::vector<int>, 4>({
-        size_t(Type->num_pins), //[0..num_pins-1]
-        size_t(Type->width),    //[0..width-1]
-        size_t(Type->height),   //[0..height-1]
-        4,                      //[0..sides-1]
+    auto& grid = g_vpr_ctx.device().grid;
+    auto result = vtr::NdMatrix<std::vector<int>, 5>({
```

Comment:
Given that we first iterate over layers and then blocks to connect the pins to track, wouldn't it be better for cache locality to assign the first dimension to the layer number? If this is hard to change, we can put a TODO comment so that if it becomes critical, people will know what they can change.